### PR TITLE
Flav/new conversation UI

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR does not implement the new action rendering, stay tune!

This PR introduces the new conversation UI ([Figma design](https://www.figma.com/design/QOxHwppG64GtPocpDhS6Y7/Product?node-id=7026-16818&t=tcXd7BFrDs4izSVa-0)). The ultimate goal is to have most of the conversation components integrated into Sparkle. However, given the current setup where all rendering parts reside in a single large file, this PR restructures the code by splitting it into proper components while also implementing the new conversation UI.

Key highlights include the new logic for grouping content fragments with the closest user message, enabling them to be rendered alongside the user message. Pagination is handled correctly – content fragments are hidden if the page does not include the user message, and they will appear once the subsequent page is fetched.

This new UI includes both a normal and a compact version. The compact version is used whenever the conversation is displayed in a modal. Additional work is required to update the background in the assistant builder, which will be addressed in an upcoming dedicated PR. Note that this PR does not implement the new action rendering – stay tuned for that!

**Before**
image

**After**
image

**Before**
image

**After**
image

## Risk

Worst case it breaks the conversation view. Pretty well tested locally. Safe to rollback.

### CODE REVIEW

**Review by JulesBelveze — APPROVED** [review]
Tested it and LGTM 💪🏼 

Random nit question though: even though all components are centered, the messages are slightly larger than the input bar, is that on purpose? (e.g. on my monitor messages are 896px and input bar is 864px)

**Comment by flvndvd** [comment]
> Tested it and LGTM 💪🏼
> 
> Random nit question though: even though all components are centered, the messages are slightly larger than the input bar, is that on purpose? (e.g. on my monitor messages are 896px and input bar is 864px)



> Tested it and LGTM 💪🏼
> 
> Random nit question though: even though all components are centered, the messages are slightly larger than the input bar, is that on purpose? (e.g. on my monitor messages are 896px and input bar is 864px)

Good catch, fixed in https://github.com/dust-tt/dust/pull/5281/commits/1b6c8bb7465e221486b09be096f948c69a5541a4.

**Review by lasryaric — COMMENTED** [review]
Left 2 comments.

File: front/components/assistant/conversation/ConversationViewer.tsx [hunk 1]
```diff
@@ -31,6 +33,12 @@ import { updateMessagePagesWithOptimisticData } from "@app/pages/w/[wId]/assista
 
 const DEFAULT_PAGE_LIMIT = 50;
 
+export type MessageWithCitationsType =
+  | AgentMessageType
+  | (UserMessageType & {
+      citations?: ContentFragmentType[];
```
[the remark this answers is not in the document]
Comment (flvndvd) [reply front/components/assistant/conversation/ConversationViewer.tsx]: Done in https://github.com/dust-tt/dust/pull/5281/commits/8f894ca091824eb1bfe411ba14b34e9575ea8e16.

File: front/components/assistant/conversation/messages/MessageContent.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import type { ContentFragmentType } from "@dust-tt/types";
+
+import { ContentFragment } from "@app/components/assistant/conversation/ContentFragment";
+import type { MessageSizeType } from "@app/components/assistant/conversation/ConversationMessage";
+import { classNames } from "@app/lib/utils";
+
+interface MessageContentProps {
+  children: React.ReactNode;
+  citations?: ContentFragmentType[];
```
[the remark this answers is not in the document]
Comment (flvndvd) [reply front/components/assistant/conversation/messages/MessageContent.tsx:9]: Added a TODO, `citations` here will be used for both content fragments on user messages and proper citations on agent message. I'll fix it in a follow up PR.

**Review by lasryaric — APPROVED** [review]
LGTM.

**Comment by spolu** [comment]
How does the content fragment work with slack thread attachement ?

**Comment by flvndvd** [comment]
> How does the content fragment work with slack thread attachement ?

Yes, it's displayed below like a regular file that the user uploads. This raises some questions about how we should display content fragments based on their types (user uploads vs slack threads). I'll check with @Duncid to see if we have any ideas.

image

**Pull request merged**